src/Tasks/XslTransformation.cs

```diff
@@ -159,13 +159,18 @@ public override bool Execute()
             // Do the transformation.
             try
             {
+                if (UseTrustedSettings)
+                {
+                    Log.LogMessageFromResources(MessageImportance.High, "XslTransform.SecuritySettingsViaUseTrustedSettings");
+                }
+
                 for (int i = 0; i < xmlinput.Count; i++)
                 {
                     using (XmlWriter xmlWriter = XmlWriter.Create(_outputPaths[i].ItemSpec, xslct.OutputSettings))
                     {
                         using (XmlReader xr = xmlinput.CreateReader(i))
                         {
-                            xslct.Transform(xr, arguments, xmlWriter);
+                            xslct.Transform(xr, arguments, xmlWriter, new XmlUrlResolver());
```

Comment:
It feels weird to me to always pass the resolver, even when not trusted. I see though that it's also always passed to the constructors of the `XslCompiledTransform`.

Should we pass it only when `UseTrustedSettings` is true?

Reply:
Since .NET Core does not allow resolving external URIs for XML by default. So, add that default resolver explicitly.  If the UseTrustedSettings is false, it won't allow the XSLT document() function. The XmlUrlResolver that resolves the document function won't work.
If the UseTrustedSettings is false and there is document, it will report the following error before resolve the document.
![image](https://user-images.githubusercontent.com/26814373/236431370-2b2d572e-428e-4a54-887e-9a34472d0dc1.png)